pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1657,16 +1658,20 @@ public PulsarAdmin getClusterPulsarAdmin(String cluster, Optional<ClusterData> c
      * @return CompletableFuture<Topic>
      * @throws RuntimeException
      */
-    protected CompletableFuture<Optional<Topic>> loadOrCreatePersistentTopic(final String topic,
+    @VisibleForTesting
+    public CompletableFuture<Optional<Topic>> loadOrCreatePersistentTopic(final String topic,
             boolean createIfMissing, Map<String, String> properties) {
         final CompletableFuture<Optional<Topic>> topicFuture = FutureUtil.createFutureWithTimeout(
                 Duration.ofSeconds(pulsar.getConfiguration().getTopicLoadTimeoutSeconds()), executor(),
                 () -> FAILED_TO_LOAD_TOPIC_TIMEOUT_EXCEPTION);
 
-        topicFuture.exceptionally(t -> {
+        topicFuture.exceptionallyAsync(e -> {
             pulsarStats.recordTopicLoadFailed();
```

Comment:
The primary change of https://github.com/apache/pulsar/pull/22860 is to remove the failed future case by case instead of removing a failed topic creation future here.

@shibd @codelipenghui  Could you also review the current PR?

Reply:
Yes, I explained all details in the PR description.

Reply:
Key point:

> But there are still some cases not considered

https://github.com/apache/pulsar/issues/24497 is the best example. https://github.com/apache/pulsar/pull/23184 is another example before.

The main concern is that if an orphan pending `PersistentTopic` could affect the new pending `PersistentTopic`, mainly due to:
- Managed ledger, which was handled in this PR
- Replicators, especially the Pulsar producer

Even if it could make the new `PersistentTopic` fail to load, it will be the same with the existing case, with better exception message (e.g. `replicator create failure`) rather than the misleading timeout exception (`Failed to load topic within timeout`).